src/main/java/appeng/tile/misc/TileCellWorkbench.java

```diff
@@ -166,26 +165,27 @@ public void onChangeInventory( final IInventory inv, final int slot, final InvOp
 
 				if( cellHasConfig )
 				{
-					for( int x = 0; x < this.config.getSizeInventory(); x++ )
+					for( int x = 0; x < this.config.getSlots(); x++ )
 					{
-						this.config.setInventorySlotContents( x, configInventory.getStackInSlot( x ) );
+						this.config.setStackInSlot( x, configInventory.getStackInSlot( x ) );
 					}
 				}
 				else
 				{
-					for( int x = 0; x < this.config.getSizeInventory(); x++ )
+					for( int x = 0; x < this.config.getSlots(); x++ )
 					{
-						configInventory.setInventorySlotContents( x, this.config.getStackInSlot( x ) );
+						configInventory.setStackInSlot( x, this.config.getStackInSlot( x ) );
 					}
 
-					configInventory.markDirty();
+					//TODO: should not be needed anymore
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1135)